osquery/tables/system/posix/yum_sources.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+void parseYumConf(const std::string& source, QueryData& results, std::string& repos_dir) {
+  // Default value
+  repos_dir = "/etc/yum.repos.d";
+
+  std::string content;
+  if (!readFile(source, content)) {
+    return;
+  }
+
+  std::string section = "";
+  for (const auto& line : osquery::split(content, "\n")) {
+    auto size = line.size();
+
+    // Skip comments.
+    if (size == 0 || line[0] == '#') {
+      continue;
+    }
+
+    if (size > 2) {
```

Comment:
This and the below `if`/`else` blocks you have get _quite_ deep in scope. Would you mind taking a crack at this `for` loop and changing your logic to `return` or `continue` early, as opposed to `if(condition) { do all the work } `?